src/SFML/Window/Android/WindowImplAndroid.cpp

```diff
@@ -70,12 +70,17 @@ m_size(mode.size)
     if (state == State::Fullscreen)
         states.fullscreen = true;
 
+    const bool shouldCreateSurface = (states.window != nullptr) && (WindowImplAndroid::singleInstance == nullptr);
+
     WindowImplAndroid::singleInstance = this;
     states.forwardEvent               = forwardEvent;
 
     // Register process event callback
     states.processEvent = processEvent;
 
+    if (shouldCreateSurface)
+        states.forwardEvent(sf::Event::FocusGained{});
```

Comment:
Does this trigger the creation?

Reply:
Yes. It is the same mechanism onNativeWindowCreated uses.

Processing the FocusGained event sets m_windowBeingCreated to true here: https://github.com/SFML/SFML/blob/387082237bda73c8cf0f47c27ae3f06c5082c6c2/src/SFML/Window/Android/WindowImplAndroid.cpp#L243

That, in turn, causes the Surface creation here: https://github.com/SFML/SFML/blob/387082237bda73c8cf0f47c27ae3f06c5082c6c2/src/SFML/Window/Android/WindowImplAndroid.cpp#L115

Reply:
For my thoughs on using opaque event types like this to achieve window creation, see: https://github.com/SFML/SFML/issues/3508